src/database/sql_database.cc

```diff
@@ -1387,6 +1389,9 @@ bool SQLDatabase::addContainer(int parentContainerId, std::string virtualPath, c
     }
     commit("addContainer");
 
+    if (cont->getMetaData(M_DATE).empty())
+        cont->addMetaData(M_DATE, to_dcDate());
+
```

Comment:
There is already a function `currentTime()` which you only need to format 
`fmt::format("{:%FT%T%z}", fmt::localtime(cont->getMTime().count()))`

To set mtime you can add `mtime = currentTime();` in constructor of  `CdsObject` and `CdsContainer`